ArduPlane/quadplane.cpp

```diff
@@ -2488,12 +2488,24 @@ int8_t QuadPlane::forward_throttle_pct(void)
     // inhibit reverse throttle and allow petrol engines with min > 0
     int8_t fwd_throttle_min = plane.have_reverse_thrust() ? 0 : plane.aparm.throttle_min;
     vel_forward.integrator = constrain_float(vel_forward.integrator, fwd_throttle_min, plane.aparm.throttle_max);
-    
-    // If we are below alt_cutoff then scale down the effect until it turns off at alt_cutoff and decay the integrator
-    float alt_cutoff = MAX(0,vel_forward_alt_cutoff);
-    float height_above_ground = plane.relative_ground_altitude(plane.g.rangefinder_landing);
-    vel_forward.last_pct = linear_interpolate(0, vel_forward.integrator,
-                                   height_above_ground, alt_cutoff, alt_cutoff+2);
+
+    if (in_vtol_auto() && is_vtol_land(plane.mission.get_current_nav_cmd().id) &&
+        (poscontrol.state == QPOS_POSITION1 || poscontrol.state == QPOS_POSITION2) &&
+        motors->get_desired_spool_state() >= AP_Motors::DESIRED_GROUND_IDLE) {
```

Comment:
I think this check is actually good/valid enough we could always apply it after this entire if/else. Forward throttle assist only makes sense/works if the vtol motors are enabled. It would add a measure of protection for rangefinder failure/terrain offsets if we made it a general check. (IE we were confident enough to turn off the VTOL props, so assist should never be responsible for spinning the forward motor up). But I'm happy to at least have it here.

Reply:
I've added this check at the top, so it never runs vfwd when vtol motors shutdown